### PR TITLE
LDAP connector - add emailSuffix config option

### PR DESCRIPTION
**Pull request by dkess**
In my organization, we don't have an LDAP mail attribute because email addresses are always based on username. This PR adds an `emailSuffix` option for this use case, where an email domain can be specified in place of a mail attribute.

### CODE REVIEW

File: connector/ldap/ldap_test.go [hunk 1]
```diff
@@ -123,6 +123,80 @@ userpassword: bar
...
+		{
+			name:      "invaliduser",
+			username:  "idontexist",
+			password:  "foo",
+			wantBadPW: true, // Want invalid password, not a query error.
+		},
+	}
+
+	runTests(t, schema, connectLDAP, c, tests)
+}
```
Comment (srenatus) [inline connector/ldap/ldap_test.go]: [nit] missing newline

File: connector/ldap/ldap_test.go [hunk 2]
```diff
@@ -123,6 +123,80 @@ userpassword: bar
 	runTests(t, schema, connectLDAP, c, tests)
 }
 
+func TestQuerySuffix(t *testing.T) {
```
Comment (srenatus) [inline connector/ldap/ldap_test.go]: Thanks for adding this test -- would you mind cutting it down a little?

I'd think two cases would be enough: the happy path, and one with the `EmailAttr` missing 🤔 Invalid password, and not-found-user don't seem required there, it's not related to the to email suffix.

[nit] Let's add Email there: 
```suggestion
func TestQueryWithEmailSuffix(t *testing.T) {
```

File: connector/ldap/ldap.go [hunk 3]
```diff
@@ -107,6 +107,9 @@ type Config struct {
 		IDAttr    string `json:"idAttr"`    // Defaults to "uid"
 		EmailAttr string `json:"emailAttr"` // Defaults to "mail"
 		NameAttr  string `json:"nameAttr"`  // No default.
+
+		// Use a username suffix instead of attribute for email.
```
Comment (srenatus) [inline connector/ldap/ldap.go]: Could you elaborate what this means a little? Maybe something along the lines of _"if this is set, the email claim of the id token will be constructed from ... etc etc"_

**Comment by dkess** [comment]
Thanks for the feedback! I believe I've addressed all the comments, let me know if anything else is necessary.

**Comment by dkess** [comment]
Done!

**Pull request merged**